clang/lib/CIR/CodeGen/TargetInfo.cpp

```diff
@@ -443,6 +470,25 @@ cir::ABIArgInfo X86_64ABIInfo::classifyArgumentType(QualType Ty,
   return cir::ABIArgInfo::getDirect(ResType);
 }
 
+// Skeleton only. Implement when used in TargetLower stage.
+cir::ABIArgInfo NVPTXABIInfo::classifyReturnType(QualType retTy) const {
+  llvm_unreachable("not yet implemented");
+}
+
+cir::ABIArgInfo NVPTXABIInfo::classifyArgumentType(QualType ty) const {
+  llvm_unreachable("not yet implemented");
+}
+
+void NVPTXABIInfo::computeInfo(CIRGenFunctionInfo &fnInfo) const {
+  if (!getCXXABI().classifyReturnType(fnInfo))
```

Comment:
See explanation in `void X86_64ABIInfo::computeInfo(CIRGenFunctionInfo &FI)` (same in AArch64ABIInfo):

> // Top level CIR has unlimited arguments and return types. Lowering for ABI
> // specific concerns should happen during a lowering phase. Assume everything
> // is direct for now.

You can basically copy the same logic for NVPTX for this whole function. In the future, whenever you actually go about implementing argument and result ABI, we'll need something like `clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/X86.cpp` for NVPTX, but not right now.